src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -66,6 +66,12 @@ public static IList<Diagnostic> GetCompileTimeConstantViolation(SyntaxBase expre
                     // The name *is* the escaped string value, so we must have an exact match.
                     return targetType.Name == sourceType.Name;
 
+                case StringLiteralType _ when sourceType is PrimitiveType:
+                    // We allow string to string literal assignment only in the case where the "AllowLooseStringAssignment" validation flag has been set.
+                    // This is to allow parameters without 'allowed' values to be assigned to fields expecting enums.
+                    // At some point we may want to consider flowing the enum type backwards to solve this more elegantly.
+                    return sourceType.Name == LanguageConstants.String.Name && sourceType.ValidationFlags.HasFlag(TypeSymbolValidationFlags.AllowLooseStringAssignment);
```

Comment:
Microoptimization nit - Could flip the order of conditions to avoid some string comparisons.

Reply:
Might as well